File: app/src/main/java/com/artmurka/artmurkaapp/Model/Pojo/ItemList/NovaPoshta/WarehousesRequest/WarehouseRequest.java

package com.artmurka.artmurkaapp.Model.Pojo.ItemList.NovaPoshta.WarehousesRequest;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class WarehouseRequest {

    @SerializedName("modelName")
    @Expose
    private String modelName;
    @SerializedName("calledMethod")
    @Expose
    private String calledMethod;
    @SerializedName("methodProperties")
    @Expose
    private MethodProperties methodProperties;
    @SerializedName("apiKey")
    @Expose
    private String apiKey;

    public String getModelName() {
        return modelName;
    }

    public void setModelName(String modelName) {
        this.modelName = modelName;
    }

    public WarehouseRequest withModelName(String modelName) {
        this.modelName = modelName;
        return this;
    }

    public String getCalledMethod() {
        return calledMethod;
    }

    public void setCalledMethod(String calledMethod) {
        this.calledMethod = calledMethod;
    }

    public WarehouseRequest withCalledMethod(String calledMethod) {
        this.calledMethod = calledMethod;
        return this;
    }

    public MethodProperties getMethodProperties() {
        return methodProperties;
    }

    public void setMethodProperties(MethodProperties methodProperties) {
        this.methodProperties = methodProperties;
    }

    public WarehouseRequest withMethodProperties(MethodProperties methodProperties) {
        this.methodProperties = methodProperties;
        return this;
    }

    public String getApiKey() {
        return apiKey;
    }

    public void setApiKey(String apiKey) {
        this.apiKey = apiKey;
    }

    public WarehouseRequest withApiKey(String apiKey) {
        this.apiKey = apiKey;
        return this;
    }

}

File: app/src/main/java/com/artmurka/artmurkaapp/data/model/pojo/itemlist/novaposhta/WarehousesRequest/WarehouseRequest.kt
package com.artmurka.artmurkaapp.data.model.pojo.itemlist.novaposhta.WarehousesRequest

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class WarehouseRequest {

    @SerializedName("modelName")
    @Expose
    var modelName: String? = null
    @SerializedName("calledMethod")
    @Expose
    var calledMethod: String? = null
    @SerializedName("methodProperties")
    @Expose
    var methodProperties: MethodProperties? = null
    @SerializedName("apiKey")
    @Expose
    var apiKey: String? = null

    fun withModelName(modelName: String): WarehouseRequest {
        this.modelName = modelName
        return this
    }

    fun withCalledMethod(calledMethod: String): WarehouseRequest {
        this.calledMethod = calledMethod
        return this
    }

    fun withMethodProperties(methodProperties: MethodProperties): WarehouseRequest {
        this.methodProperties = methodProperties
        return this
    }

    fun withApiKey(apiKey: String): WarehouseRequest {
        this.apiKey = apiKey
        return this
    }

}
